File: 1.0/src/c/cz/mg/vulkan/PFN_vkWaitForFences.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkWaitForFences_call(JNIEnv* env, jclass jc, jlong address, jlong device, jlong fenceCount, jlong pFences, jlong waitAll, jlong timeout, jlong rval){
    (void)env;
    (void)jc;
    PFN_vkWaitForFences f = (PFN_vkWaitForFences)jniLongToFunctionPointer(address);
    VkResult* rvalAddress = jniLongToPointer(rval);
    *rvalAddress = f(
        *((VkDevice*)jniLongToPointer(device)),
        *((uint32_t*)jniLongToPointer(fenceCount)),
        ((VkFence*)jniLongToPointer(pFences)),
        *((VkBool32*)jniLongToPointer(waitAll)),
        *((uint64_t*)jniLongToPointer(timeout))
    );
}

File: 1.0/src/cz/mg/vulkan/PFNvkWaitForFences.java
package cz.mg.vulkan;

public class PFNvkWaitForFences extends VkFunctionPointer {
    public PFNvkWaitForFences() {
    }

    protected PFNvkWaitForFences(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkWaitForFences(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkWaitForFences(long value) {
        setValue(value);
    }

    public PFNvkWaitForFences(VkInstance instance) {
        super(instance, new VkString("vkWaitForFences"));
    }

    public void call(VkDevice device, VkUInt32 fenceCount, VkFence pFences, VkBool32 waitAll, VkUInt64 timeout, VkResult rval){
        callNative(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddressNative(), fenceCount != null ? fenceCount.getVkAddress() : VkPointer.getNullAddressNative(), pFences != null ? pFences.getVkAddress() : VkPointer.NULL, waitAll != null ? waitAll.getVkAddress() : VkPointer.getNullAddressNative(), timeout != null ? timeout.getVkAddress() : VkPointer.getNullAddressNative(), rval != null ? rval.getVkAddress() : VkPointer.getSinkAddressNative());
    }

    protected static native void callNative(long vkaddress, long device, long fenceCount, long pFences, long waitAll, long timeout, long rval);


    public int call(VkDevice device, int fenceCount, VkFence pFences, int waitAll, long timeout){
        return callSimplifiedNative(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddressNative(), fenceCount, pFences != null ? pFences.getVkAddress() : VkPointer.NULL, waitAll, timeout);
    }

    protected static native int callSimplifiedNative(long vkaddress, long device, int fenceCount, long pFences, int waitAll, long timeout);

}
